pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -169,6 +169,42 @@ func (e *cloudWatchExecutor) CallResource(ctx context.Context, req *backend.Call
 	return e.resourceHandler.CallResource(ctx, req, sender)
 }
 
+func (e *cloudWatchExecutor) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	status := backend.HealthStatusOk
+	metricsTest := "Successfully queried the CloudWatch metrics API."
+	logsTest := "Successfully queried the CloudWatch logs API."
+
+	// use billing metrics for metrics test
+	namespace := "AWS/Billing"
+	metric := "EstimatedCharges"
+	params := &cloudwatch.ListMetricsInput{
+		Namespace:  &namespace,
+		MetricName: &metric,
+	}
+	_, err := e.listMetrics(req.PluginContext, defaultRegion, params)
```

Comment:
nit: just code style but I would split this function in two `CheckHealthMetrics` and `CheckHealtLogs` (it would be easier to test individually too if needed)

Reply:
done